grpc.go

```diff
@@ -310,6 +356,24 @@ func (a *agentGRPC) runProcess(cid string, agentProcess *pb.Process) (pid int, e
 	return pid, nil
 }
 
+func (a *agentGRPC) updateContainerConfig(spec *specs.Spec, config *configs.Config) error {
+	// Update shared PID namespace.
+	for idx, ns := range config.Namespaces {
+		if ns.Type == configs.NEWPID {
+			// In case the path is empty because we don't expect
+			// the containers to share the same PID namespace, a
+			// new PID ns is going to be created.
+			config.Namespaces[idx].Path = a.sandbox.sharedPidNs.path
```

Comment:
I feel the comment is a little weird: If the Path is empty and `Type == NEWPID`, then this means the container wants to leave on its own PID namespace. But the code sets it to `a.sandbox.sharedPidNs.path`. AFAIU there are a few assumptions here:

1) If `ns.Path` is empty and the pod does not want shared PID ns, `a.sandbox.sharedPidNs.path` is going to be empty, and this is a NO-OP.
2) If `ns.Path` is NOT empty, then we're overwriting is with the shared PID ns, i.e. shared PID ns takes precedence over a specific PID ns the container may want to join. AFAIK this is not documented anywhere.
3) If the shared PID ns does not exist at that point, libcontainer will create it.

Could you please add more comments to make those assumptions more explicit ?

Reply:
I have added more comments. Also, a follow up PR should actually take care of the case where the spec gives us some paths for the namespace types defined. Right now, we ignore any ns paths since we don't know how to handle it.
We can make something simple but the question is how k8s would handle a shared ns case through the spec ? I mean that we need to decide from where the information should come in case of shared ns.
For instance, if k8s uses properly the spec to share the same ns between containers, then we only need to rely on the spec, instead of passing a specific parameter when creating the sandbox (as @laijs proposed last week IIRC).

Reply:
@sboeuf what about the case that no `ns.Type == configs.NEWPID` set in spce? Which pidns does the container live in when libcontainer creates it?

Reply:
Well in that case, the container will share the same PID ns than the agent. I don't think we want this case to happen, and that's why I was asking if we should consider some default values that we would apply from the agent every time.

Reply:
Yeah, we don't want it to happen. pidns should either be the pause process one, or containers live on their own. I think the default value is controlled by `SandboxPidns` field in CreateSandbox() request.

Reply:
@sboeuf Agree with @laijs  and you here: We don't want the container to live in the agent pidns. So depending on the `SandboxPidns` value, the default should be the pause process one or we create a new pidns for each container.

Reply:
I would prefer that by default, if we have no `ns.Type == configs.NEWPID` from the list of namespaces provided by the spec, we actually add one from the agent. And then, we apply the `sandboxPidns`. This means that we make sure the container will end up in its own PID ns, or in the pause container one if `sandboxPidns` is not empty. Here, we would be adding a security in case the runtime does not provide this `NEWPID` type.
Are you all okay with that ? @sameo @bergwolf @laijs 
The other solution would be to throw an error in case the `NEWPID` type is not provided by the spec.

Reply:
Your proposal sounds good to me @sboeuf. I would prefer that agent handles it rather than throwing errors when `NEWPID` is not set.

Reply:
@bergwolf ok great, I'll add this to the code then!